soc/ti/k3/soc.yml

```diff
@@ -10,3 +10,6 @@ family:
     - name: am6442
       cpuclusters:
       - name: m4
+    - name: j721e
+      cpuclusters:
+      - name: main/r5f0_0
```

Comment:
blocking comment.

Looking at the datasheet for j721e I do see the use of domains, and that there are two domains, main and mcu, each having a number of cores.

So I understand the wish / logic in representing this by exposing the domains in the cluster name, unfortunately the `/` is also used as separator for the board qualifier and thus i'm concerned how underlying implementation will handle an extra `/` inside the CPU cluster. Especially as there can be multiple levels of board variants, and well as a future enhancements, see https://github.com/zephyrproject-rtos/zephyr/issues/69546, could suddenly be impacted.

Currently my expectation is that the `r5f0_0` probably will be treated as a board variant and thus work, but before approving I would like to take an extra look before deciding if this construct can be allowed, or if an extra field should be introduced for CPU cluster.

Alternative could be to name the cpu cluster as: `main_rf50_0`

Reply:
So the concern is around how to parse out which part of the following is the board:

`-b <board>/<board variant>/<soc>/<domain>/<core>`

when there are variants/revisions of a board?

Not sure why the board variant/revision would be put in the target string that way to begin with. It doesn't follow the normal pattern of the `/` separator meaning a nesting (i.e. contains one or more of x). Boards contain a number of SoCs, they do not *contain* a number of revisions.. Our SoCs do contain a number of domains which contain a number of cores, that matches perfectly with the existing convention.

I'd suggest then board revision simply be backed into the board string (e.g. board_deluxe_edition_revc/soc/etc..). That way, should some some variant of a board contain a different set of SoCs than the original, it could be represented correctly.

If the `/` does cause end up causing some issue, and we do not want to add sub-fields to CPU cluster, then main_rf50_0` is also fine with me if needed also.

Reply:
@tejlmand , gentle ping. Any comment on the above? This seems to be the last outstanding item on this PR

Reply:
To be safe I've just changed the name to `main_rf50_0`, should resolve your concern.